src/react/components/StyleForm.jsx

```diff
@@ -0,0 +1,348 @@
+
+        this.ref = React.createRef();
+
+        // initialize state with these initialization parameters to show preview correctly
+        this.state = this.props.styleSettings;
+        this.state.format = 'point'; // Add format manually which doesnt come with style JSON
+
+        this._populateWithStyle = (format) => {
+            if (this.props.styleList) {
+                let currentStyle = this.props.styleList.find(option => option.value == format);
+                this.setState({
+                    ...currentStyle,
+                    format: currentStyle.format
+                });
+    
+                this.ref.current.setFieldsValue(currentStyle); // Populate fields -- FIX ME
+            }
+        }
+
+        this.styleInputCallback = (event) => {
+            this.setState({
+                [event.target.id || event.target.name]: {
+                    ...this.state[event.target.id || event.target.name],
+                    color: event.target.value 
+                }
+            });
+        };
+
+        this.changeTab = (event) => this.setState({ [event.target.name]: event.target.value });
+
+        this.sizeControlCallback = (value) => this.setState({ 'strokeWidth': value, 'size': value });
+
+        /**
+         * 
+         * @param {String} targetString - target parameter in object provided in full dot notation 
+         * @param {String|Number} value - value to be set 
+         */
+        this.updateState = (targetString, value) => {
+            const firstTarget = targetString.substr(0, targetString.indexOf('.'));
+            let currentTarget = this.state;
+            currentTarget = this.parseThru(currentTarget, targetString, value);
+
+            this.setState({
+                [firstTarget]: {
+                    ...this.state[firstTarget],
+                    currentTarget
+                }
+            });
+        };
+
+        /**
+         * 
+         * @param {Object} targetObject - target object to be parsed through
+         * @param {String} targetString - target parameter in object provided in full dot notation 
+         * @param {String|Number} value - value to be set
+         */
+        this.parseThru = (targetObject, targetString, value) => {
+            if (typeof targetString === 'string') {
+                return this.parseThru(targetObject, targetString.split('.'), value); // first cycle of recursion converts targetString into array
+            } else if (targetString.length == 1 && value !== undefined) {
+                return targetObject[targetString[0]] = value; // We reach end of the recursion
+            } else if (targetString.length === 0) {
+                return targetObject; // target is already on level 0 so no need for recursion
+            } else {
+                return this.parseThru(targetObject[targetString[0]], targetString.slice(1), value); // recursive call and remove first element from array
+            }
+        }
+    }
```

Comment:
The tab selection can probably be handled as a state of the form component but all of this other stuff should be moved to a controller/uihandler type file. Similar to the one in admin-layereditor for example: https://github.com/oskariorg/oskari-frontend/blob/master/bundles/admin/admin-layereditor/view/AdminLayerForm/AdminLayerFormHandler.js

Reply:
FormHandler added to `src/react/util/extras/StyleFormHandler.js`. Should this be somewhere else?